### ISSUE TITLE
搭建模型中exe.run()出现数据类型错误

### ISSUE THREAD

**Issue by BlackSpritee** [PaddlePaddle/Paddle#40969]
环境是paddle1.8 pgl1.2.1
报错如下
_as_lodtensor
    tensor.set(data, place)
paddle.fluid.core_avx.EnforceNotMet: 

--------------------------------------------
C++ Call Stacks (More useful to developers):
--------------------------------------------
0   std::string paddle::platform::GetTraceBackString<std::string>(std::string&&, char const*, int)
1   paddle::platform::EnforceNotMet::EnforceNotMet(paddle::platform::ErrorSummary const&, char const*, int)

----------------------
Error Message Summary:
----------------------
Error: Incompatible data or style type: tensor.set() supports bool, float16, float32, float64, int8, int16, int32, int64 and uint8, uint16, but got object! at (/paddle/paddle/fluid/pybind/tensor_py.h:242)

feed_dict添加一类数据后出现以上错误，相对于feed_dict中的其他参数，这类数据与其他参数区别的数据类型为list，请问是这个原因吗？或者说这个错误可能的情况有哪些？

**Comment by yeyupiaoling** [comment]
你这个list转成numpy试试？

**Comment by BlackSpritee** [comment]
> 你这个list转成numpy试试？

feed到模型的数据就是用的numpy类型，还是报上述错误

**Comment by yeyupiaoling** [comment]
我看看你的exe.run是怎么传的？

**Comment by BlackSpritee** [comment]
> 我看看你的exe.run是怎么传的？

好的，大致的训练过程如下
def train(config):
    num_nodes, num_edges, edges, entity_set, ent2id, rel2id \
        = load_graph_data(config)
    print(num_nodes)
    print(num_edges)
    config.num_class = len(rel2id)
    print(rel2id)
    train_dataset = load_train_test_data(config.train_data_path, ent2id, rel2id)
    node_feature, args.cls_id, sentence_ids = dump_node_feat(args, ent2id ,train_dataset[3])
    for i,sentence_id in enumerate(sentence_ids):
        train_dataset[3][i]=sentence_id[0]

    feed_dict = {
    "num_nodes": np.array([num_nodes]),
    "num_edges": np.array([num_edges]),
    "edges": np.array(edges),
    "term_ids": np.array(node_feature),
    }

    def train_reader():
        for index in range(len(train_dataset.label)):
            yield [train_dataset.e1_index[index], train_dataset.e2_index[index],
                   train_dataset.label[index],train_dataset.sentence[index]]

    train_batch_dataset = F.io.batch(train_reader, batch_size=config.batch_size)

    num_nodes = L.data("num_nodes", [1], False, 'int64')
    num_edges = L.data("num_edges", [1], False, 'int64')
    edges = L.data("edges", [-1, 2], False, 'int64')
    node_feat = L.data("term_ids", [-1,config.node_max_seqlen], False, 'int64')
    gw = BatchGraphWrapper(num_nodes, num_edges, edges, {"term_ids": node_feat})

    place = F.CUDAPlace(config.gpu)
    exe = F.Executor(place)
    start_program = F.default_startup_program()
    train_program = F.default_main_program()
    # test_program = F.default_main_program().clone(for_test=True) #!!!!

    inputs_e1 = L.data("inputs_e1", [-1], False, 'int64')
    inputs_e2 = L.data("inputs_e2", [-1], False, 'int64')
    real_label = F.data("real_label", [None,1], 'int64')
    sentence_ids = L.data("sentence_ids", [-1], False,'int64')
    feed_dataset = (inputs_e1, inputs_e2, real_label, sentence_ids)
    loss, acc, pred = build_model(feed_dataset, gw, config=args)

    exe.run(start_program)
    for epoch in range(config.epoch):
        Loss_all=0
        Acc_all=0
        for i,train_iter in enumerate(train_batch_dataset()):
            # train—iter [3*B]->3*[B]
            feed_dict["inputs_e1"] = np.array(train_iter,dtype=object)[:, 0]
            feed_dict["inputs_e2"] = np.array(train_iter,dtype=object)[:, 1]
            feed_dict["real_label"] = np.array(train_iter,dtype=object)[:, 2].reshape([len(np.array(train_iter,dtype=object)[:, 0]), 1])
            feed_dict["sentence_ids"] = np.array(train_iter, dtype=object)[:, 3]

            train_loss, train_acc, train_pred = exe.run(train_program,feed=feed_dict, fetch_list=[loss, acc, pred])
            Loss_all+=train_loss[0]
            Acc_all+=train_acc[0]

        print("Epoch", epoch, "Acc ", Acc_all/(i+1), "loss ",Loss_all/(i+1) , "pre ", train_pred)

**Comment by yeyupiaoling** [comment]
你检查看你输入得数据是不是list，报错提示你输入的是一个object。你打印看看。

**Comment by BlackSpritee** [comment]
> 你检查看你输入得数据是不是list，报错提示你输入的是一个object。你打印看看。
打印的结果如下
name: " sentence_index"
type {
  type: LOD_TENSOR
  lod_tensor {
    tensor {
      data_type: INT64
      dims: -1
    }
    lod_level: 0
  }
}
persistable: false

name: "sentence_feat"
type {
  type: LOD_TENSOR
  lod_tensor {
    tensor {
      data_type: INT64
      dims: -1
      dims: 128
    }
    lod_level: 0
  }
}
persistable: false

**Comment by BlackSpritee** [comment]
> 你检查看你输入得数据是不是list，报错提示你输入的是一个object。你打印看看。
打印结果如上，sentence_feat并不是list类型，请问需要如何检查，谢谢~

**Comment by yeyupiaoling** [comment]
我看你打印的是层，我想让你打印的是这里的值

```
feed_dict = {
"num_nodes": np.array([num_nodes]),
"num_edges": np.array([num_edges]),
"edges": np.array(edges),
"term_ids": np.array(node_feature),
}
```

**Comment by BlackSpritee** [comment]
在feed_dict之前打印结果如下，在feed_dict之后报错在print之前
     print(np.array(sentence_ids))
     feed_dict = {
    "num_nodes": np.array([num_nodes]),
    "num_edges": np.array([num_edges]),
    "edges": np.array(edges),
    "term_ids": np.array(node_feature),
    "sentence_feat": np.array(sentence_ids),
    }
[[  375   325  2226 ... 32067  2011     5]
 [   20    15 45160 ...     0     0     0]
 [ 1315  1037     7 ...     0     0     0]
 ...
 [  317  1224   585 ...     0     0     0]
 [   30 43532  2558 ...     0     0     0]
 [41500  6076 22677 ...     0     0     0]]

**Comment by yeyupiaoling** [comment]
看数据好像是正常的，这就看不出是哪里问题了

**Comment by BlackSpritee** [comment]
> 看数据好像是正常的，这就看不出是哪里问题了

好的，感谢你的回复！

**Comment by BlackSpritee** [comment]
> 看数据好像是正常的，这就看不出是哪里问题了

解决了，是numpy默认为object类型的问题dtype=object，修改为int64就可以了。
另外请问一下ernie-tiny预训练模型支持英文吗？

**Comment by yeyupiaoling** [comment]
默认居然是object的，它应该会根据数据自动转换才对，例如你那个它应该会自动指定为int64的，不过决解就行。

ernie是支持英文的：

https://github.com/PaddlePaddle/ERNIE#3-%E4%B8%8B%E8%BD%BD%E9%A2%84%E8%AE%AD%E7%BB%83%E6%A8%A1%E5%9E%8B%E5%8F%AF%E9%80%89
